src/test/java/uk/gov/justice/hmpps/datacompliance/repository/jpa/repository/ManualRetentionRepositoryTest.java

```diff
@@ -0,0 +1,91 @@
+package uk.gov.justice.hmpps.datacompliance.repository.jpa.repository;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.transaction.TestTransaction;
+import uk.gov.justice.hmpps.datacompliance.repository.jpa.model.retention.ManualRetention;
+import uk.gov.justice.hmpps.datacompliance.repository.jpa.model.retention.ManualRetentionReason;
+
+import javax.transaction.Transactional;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.justice.hmpps.datacompliance.repository.jpa.model.retention.RetentionReasonCode.Code.HIGH_PROFILE;
+
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles("test")
+@SpringBootTest
+@Transactional
+@DirtiesContext(classMode = ClassMode.AFTER_EACH_TEST_METHOD)
+class ManualRetentionRepositoryTest {
+
+    private static final LocalDateTime NOW = LocalDateTime.now();
+
+    @Autowired
+    private ManualRetentionRepository manualRetentionRepository;
+
+    @Autowired
+    private RetentionReasonCodeRepository reasonCodeRepository;
+
+    @Test
+    @Sql("retention_reason_code.sql")
+    void persistManualRetentionAndRetrieveById() {
+
+        final var manualRetention = ManualRetention.builder()
+                .offenderNo("A1234BC")
+                .staffId(1234L)
+                .retentionDateTime(NOW)
+                .build();
+
+        manualRetention.addManualRetentionReason(ManualRetentionReason.builder()
+                .retentionReasonCodeId(reasonCodeRepository.findById(HIGH_PROFILE).orElseThrow())
+                .reasonDetails("High profile for some reason")
+                .build());
+
+        manualRetentionRepository.save(manualRetention);
+
+        TestTransaction.flagForCommit();
```

Comment:
So I was looking for an alternative to using `TestTransaction`.  I couldn't really find any good articles on it.  The approach I did try though was autowiring in the `TestEntityManager` and than doing a `entityManager.flush()` when I needed to push changes to the DB.  In essence at this point it becomes 1 line instead of 3.  Still not sure though....

Reply:
Yes it does seem a bit verbose doing this, I have wondered about the benefit of proving the transaction (what extra does it give us apart from showing that HSQLDB can commit transactions?)